### ISSUE TITLE
How to set username:password for privat gitlab repository?

### ISSUE THREAD

**Issue by Nowi5** [lorisleiva/laravel-deployer#73]
Hi,

I really like this package and the howto's you have created. Besides the fact that I need to trust my gitlab host (e.g. gitlab.com) to work with my private SSH Key, I run into another challenge: My gitlab repositories are private so the deploy fails:

> In Client.php line 99:                                                                                                                              
>     The command "cd /var/customers/XYZ.com/ && (/usr/bin/git clone   --recursive -q https://gitlab.com/XYZ/test.git /var/customers/XYZ.com//releases/1 2>&1)" failed.

![grafik](https://user-images.githubusercontent.com/27913135/49704252-fe3eb980-fc10-11e8-971f-d1ac943e129a.png)

When I tested in my own docker, I had the same issue, so running the commands one by one by myself shows that /usr/bin/git clone asks for username and password.
![grafik](https://user-images.githubusercontent.com/27913135/49704271-2f1eee80-fc11-11e8-94ae-338029a69040.png)

How can I add this to my repository settings? 

Thank you for your help.

**Comment by Nowi5** [comment]
And here is the answer; 
Set the repository to the related link which includes an auth token e.g. **https://oauth2:ACCESS_TOKEN@somegitlab.com/vendor/package.git**

here you get the personal access token: https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html

Here some further help: https://stackoverflow.com/questions/25409700/using-gitlab-token-to-clone-without-authentication. 
The version with gitlab-ci-token did not worked for me.

**Comment by lorisleiva** [comment]
👋 Hi there

Thanks, I'm glad you like this package. 🙂 

I've never had to deal with this issue so I'm pleased to hear you've find a way to make it work. Thank you for sharing the answer for potential future readers. 🙏

**Comment by simonbuehler** [comment]
had the same issue and thanks for the oauth solution, save my day, but commiting a token in deploy.php to git is not ideal, how did you solve that?

**Comment by stevepop** [comment]
I am also interested in an answer to the question above. I recently started using laravel-deploy but my repos are private on Github. Deploying with repository set with the token works but I can't commit this to the repository as it is not ideal. Is there a way to dynamically read this into deploy.php something like below?
```

    'options' => [
        'application' => env('APP_NAME', 'Laravel'),
        'repository' => 'https://{{ token }}@github.com/you/repo.git',
    ],
```

**Comment by lorisleiva** [comment]
Hi there 👋 

I have just documented something that could solve your problem:
https://github.com/lorisleiva/laravel-deployer/blob/master/docs/how-to-cli-variable.md

In your case, you could have something like this:

```php
// config/deploy.php

'options' => [
    'token' => '',
    'repository' => 'https://{{ token }}@github.com/you/repo.git',
    // ...
],
```

And then you can provide the token via a private environment variable

```sh
php artisan deploy -o token=$(echo MY_PRIVATE_TOKEN)
```

I hope this helps. 🍀

**Comment by stevepop** [comment]
Hi @lorisleiva, thanks for this. I tried the above but I got the following errors;
```
➜  intrinsic git:(new-admin-theme) ✗ php artisan deploy -o token=$(echo GITHUB_TOKEN)

   ErrorException  : Array to string conversion

  at /Users/stevepop/Sites/intrinsic/vendor/laravel/framework/src/Illuminate/Support/Collection.php:538
    534|      * @return static
    535|      */
    536|     public function intersect($items)
    537|     {
  > 538|         return new static(array_intersect($this->items, $this->getArrayableItems($items)));
    539|     }
    540|
    541|     /**
    542|      * Intersect the collection with the given items by key.

  Exception trace:

  1   array_intersect([])
      /Users/stevepop/Sites/intrinsic/vendor/laravel/framework/src/Illuminate/Support/Collection.php:538

  2   Illuminate\Support\Collection::intersect()
      /Users/stevepop/Sites/intrinsic/vendor/lorisleiva/laravel-deployer/src/LaravelDeployer/Commands/BaseCommand.php:52

```
This was my options;
```
 'options' => [
        'application' => env('APP_NAME', 'Intrinsic'),
        'token' => '',
        'repository' => 'https://{{ token }}@github.com/stevepop/intrinsic.git',
        'php_fpm_service' => 'php7.2-fpm',
    ],
```
And I ran the deploy as follows;
```
php artisan deploy -o token=$(echo GITHUB_TOKEN)
```

**Comment by lorisleiva** [comment]
Thank you for raising this. It was a bug that I, unfortunately, introduced when upgrading the package to Laravel 7. This should now be fixed from release [`v0.3.1`](https://github.com/lorisleiva/laravel-deployer/releases/tag/v0.3.1).
